### PR TITLE
chore(ipld): extract byzantine error and constructor for it into a separate file

### PR DESCRIPTION
**Pull request by Wondertan**
* It does make sense to keep ErrByzantine in ipld/get.go
* Not to bloat Retriever

### CODE REVIEW

File: ipld/retriever_byzantine.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		plugin.MustCidFromNamespacedSha256(root),
+		errShares,
+	)
+	if err != nil {
+		// Fatal as rsmt2d proved that error is byzantine,
+		// but we cannot properly collect the proof,
+		// so verification will fail and thus services won't be stopped
+		// while we still have to stop them.
+		// TODO(@Wondertan): Find a better way to handle
+		log.Fatalw("getting proof for ErrByzantine", "err", err)
```
Comment (vgonkivs) [inline ipld/retriever_byzantine.go:69]: may be we can panic here?

Comment (Wondertan) [reply ipld/retriever_byzantine.go:69]: It is basically the same

Comment (Wondertan) [reply ipld/retriever_byzantine.go:69]: Where?

Comment (Wondertan) [reply ipld/retriever_byzantine.go:69]: Panic is dev oriented way of exiting the app. Log fatal is user-oriented and does not spit with stack traces into the user

Comment (vgonkivs) [reply ipld/retriever_byzantine.go:69]: Actually, we are using `log.Fatal` only once.

Comment (vgonkivs) [reply ipld/retriever_byzantine.go:69]: But it's up to you

**Review by renaynay — APPROVED** [review]

**Comment by renaynay** [comment]
@Wondertan maybe rebase and re-run ci

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**